Global default configuration file

achilleas-k opened the pull request:
### Default configuration file
#### Repeat of #72 — Updated and removed date formatting changes

Created a default configuration file `default.conf` in the root of the code directory which holds the default configuration options (identical to the built-in values). Users may be directed to this file as a formatting guide for writing their own configuration file. `configargparse` loads the file with lower priority than user configuration file, but higher than built-in values.

Configuration priority is:

1. Command line arguments
2. User configuration file `<user_config_dir>/hangups.conf`
3. Default configuration file `<hangups root>/default.conf`
4. Built-in values

tdryer commented on hangups/default.conf:
```diff
@@ -0,0 +1,11 @@
+key-quit: ctrl e
+key-menu: ctrl n
+key-up: k
+key-down: j
```
Should this file be in the root of the repo instead? Usually that's going to be the working directory, which is where hangups will look for it.

achilleas-k replied:
Oh. I thought the root working directory of the application, after installation, would be the hangups directory (under the root of the repo directory). At least that's what it looks like on my system, e.g., the `ui` directory is at `/usr/lib/python3.5/site-packages/hangups/ui/`. If we put default.conf in the root of the repository, where would it end up after installation?

tdryer replied:
`__main__.py` is currently loading `default.conf` from whatever the current directory is when hangups is launched, eg. if you're in `$HOME/hangups`, it will look for `$HOME/hangups/default.conf`

Would putting it in site-packages along with the code be useful to anyone? It seems like that wouldn't be discoverable.

achilleas-k replied:
I see your point. I was thinking it could act as a global default, but that's not useful at all. I'll move it to the root of the repo. I'll also add the default datetime formats, since they've now been merged into main.

tdryer commented:
Should we rename `default.conf` to `hangups.conf` for consistency with the user config?

It would also be nice to have the tests verify that the the new config file says in sync with the defaults defined in `__main__.py`. That could be done later thought.

achilleas-k commented:
I've renamed `default.conf` to `hangups.conf` and changed the appropriate loading line in `__main__.py`. As for the tests, I guess that can be taken care of later, yes.

tdryer commented:
Great! Thanks.
